assets/javascripts/test_result.js

```diff
@@ -497,27 +497,30 @@ function loadTabPanelElement(tabName, tabConfig) {
     return false;
   }
   tabConfig.panelElement = tabPanelElement; // for easier access in custom renderers
-  $.ajax({
-    url: ajaxUrl,
-    method: 'GET',
-    success: function (response) {
+  fetch(ajaxUrl, {method: 'GET'})
+    .then(response => {
+      if (!response.ok) throw `Server returned ${response.status}: ${response.statusText}`;
+      if (response.headers.get('Content-Type').includes('application/json')) return response.json();
+      return response.text();
+    })
+    .then(response => {
       const customRenderer = tabConfig.renderContents;
       if (customRenderer) {
         return customRenderer.call(tabConfig, response);
       }
       tabPanelElement.innerHTML = response;
-    },
-    error: function (xhr, ajaxOptions, thrownError) {
+    })
+    .catch(error => {
+      console.error(error);
       const customRenderer = tabConfig.renderError;
       if (customRenderer) {
-        return customRenderer.call(tabConfig, xhr, ajaxOptions, thrownError);
+        return customRenderer.call(tabConfig, error);
       }
       tabPanelElement.innerHTML = '';
       tabPanelElement.appendChild(
-        document.createTextNode('Unable to load ' + (tabConfig.descriptiveName || tabName) + '.')
+        document.createTextNode('Unable to load ' + (tabConfig.descriptiveName || tabName) + `: ${error}`)
```

Comment:
It would be nicer to just use one quoting style at a time:
```suggestion
        document.createTextNode(`Unable to load ${tabConfig.descriptiveName || tabName}: ${error}`)
```